metrics_collector/instance_manager_collector.go

```diff
@@ -188,3 +207,34 @@ func (imc *InstanceManagerCollector) collectRequestValues(ch chan<- prometheus.M
 		ch <- prometheus.MustNewConstMetric(imc.memoryRequestMetric.Desc, imc.memoryRequestMetric.Type, requestMemoryBytes, imc.currentNodeID, pod.GetName(), instanceManagerType)
 	}
 }
+
+func (imc *InstanceManagerCollector) collectGrpcConnection(ch chan<- prometheus.Metric) {
+	defer func() {
+		if err := recover(); err != nil {
+			imc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	podList, err := imc.ds.ListPodsRO(imc.namespace)
+	if err != nil {
+		imc.logger.WithError(err).Warn("error during scrape")
+		return
+	}
+
+	for _, pod := range podList {
```

Comment:
Any reason you prefer to list pods rather than IM CRs? Or you can list both.
One benefit of listing/checking the CRs is that you can verify the counter backing to 0 when the pod is done.

Reply:
Updated.